dask/utils.py

```diff
@@ -623,11 +623,12 @@ def dispatch(self, cls):
             # Is a lazy registration function present?
             toplevel, _, _ = cls2.__module__.partition(".")
             try:
-                register = self._lazy.pop(toplevel)
+                register = self._lazy[toplevel]
             except KeyError:
                 pass
             else:
                 register()
+                self._lazy.pop(toplevel)
```

Comment:
A first test failed with a `KeyError` here. In hindsight, that is unsurprising since we hypothesize that the whole problem here is different threads manipulating the `_lazy` dict, so I started another test with the following modification:
```suggestion
                self._lazy.pop(toplevel, None)
```

Reply:
The test with this modification seems to work :tada: